server/src/updaters/relationship-updaters.js

```diff
@@ -0,0 +1,202 @@
+// @flow
+
+import type { Viewer } from '../session/viewer';
+
+import invariant from 'invariant';
+
+import {
+  type RelationshipStatus,
+  relationshipStatuses,
+  type RelationshipRequest,
+} from 'lib/types/relationship-types';
+import { ServerError } from 'lib/utils/errors';
+
+import { dbQuery, SQL } from '../database';
+
+async function updateRelationship(
+  viewer: Viewer,
+  request: RelationshipRequest,
+) {
+  const { userID, status } = request;
+
+  if (viewer.id === userID) {
+    throw new ServerError('invalid_user');
+  }
+
+  const userQuery = SQL`SELECT 1 FROM users WHERE id = ${userID} LIMIT 1`;
+  const [users] = await dbQuery(userQuery);
+
+  if (!users.length) {
+    throw new ServerError('user_does_not_exist');
+  }
+
+  // friend request
+  if (status === relationshipStatuses.PENDING_FRIEND) {
+    const [relationships] = await fetchRelationshipQuery(viewer.id, userID);
```

Comment:
1. 👍
2. Can you remind me what is the difference between `id` and `userID` in the `viewer` object, please? I remember it's something about the anonymous user. Sorry, I didn't have enough context where you explained it to me :( 
3. I should replace all `viewer.id` expressions with `viewer.userID` in this updater function, right?

Reply:
Yeah, it's purely for anonymous cookies. All viewers have an ID - if it is an anonymous viewer it is the cookie ID, and if it is a logged-in user then it is the user ID.

Since you added the `isLoggedIn` check at the top, it is exactly guaranteed to be the exact same ID. I just suggested to use `userID` instead of `id` just to make the code more clear that it only works with a user. If you call `viewer.userID` on an anonymous viewer it should throw